test/export/test_export.py

```diff
@@ -9026,6 +9026,28 @@ def test_dynamic_shapes_serdes_user_errors(self):
             }
             _load_dynamic_shapes(spec, from_dict=True)
 
+    def test_attention(self):
+        class Foo(torch.nn.Module):
+            def __init__(self) -> None:
+                super().__init__()
+                self.embed_dim = 768
+                self.num_heads = 12
+                self.dropout = 0.0
+                self.batch_first = True
+                self.self_attention = torch.nn.MultiheadAttention(
+                    self.embed_dim,
+                    self.num_heads,
+                    dropout=self.dropout,
+                    batch_first=self.batch_first,
+                )
+
+            def forward(self, input1: torch.Tensor):
+                x, _ = self.self_attention(input1, input1, input1, need_weights=False)
+                return x
+
+        inps = (torch.randn(1, 224, 768, device="cpu"),)
+        export(Foo(), inps)
```

Comment:
would this test have failed prior to this change? (IIUC, the main change in this test is that it will "fix" any CIA decomps that branch on cpu/cuda device, because previously we would execute them with meta tensors)

Reply:
Added a comment explaining the old behaviour